test/test_agent_jax.py

```diff
@@ -490,6 +494,205 @@ def test_agent_with_A_learning_requires_pA(self):
 
         with self.assertRaises(AssertionError):
             Agent(A=A, B=B, learn_A=True)
+
+    def test_agent_construction_jittable(self):
+        """
+        Test that the constructor call of the Agent class is jittable
+        """
+        num_obs = [3, 4]
+        num_states = [4, 5, 6]
+        num_controls = [2, 3, 1]
+        A_dependencies = [[0], [0, 1, 2]]
+        B_dependencies = [[0], [1], [2]]
+        batch_size = 2
+
+        A = utils.random_A_matrix(
+            num_obs, num_states, A_factor_list=A_dependencies
+        )
+        B = utils.random_B_matrix(
+            num_states, num_controls, B_factor_list=B_dependencies
+        )
+        pA = utils.dirichlet_like(A, scale=1.0)
+        pB = utils.dirichlet_like(B, scale=1.0)
+        D = utils.random_single_categorical(num_states)
+
+        def _broadcast(arr_list):
+            return [
+                jnp.broadcast_to(jnp.array(arr), (batch_size,) + arr.shape)
+                for arr in arr_list
+            ]
+
+        A_batched = _broadcast(A)
+        B_batched = _broadcast(B)
+        pA_batched = _broadcast(pA)
+        pB_batched = _broadcast(pB)
+        D_batched = _broadcast(D)
+
+        def construct_agent(A, B, pA, pB, D):
+            """ Constructs simple wrapping function that just builds an Agent """
+
+            agent = Agent(
+                A,
+                B,
+                D=D,
+                A_dependencies=A_dependencies,
+                B_dependencies=B_dependencies,
+                num_controls=num_controls,
+                batch_size=batch_size,
+                learn_A=True,
+                learn_B=True,
+                pA=pA,
+                pB=pB,
+            )
+
+            return agent
+        
+        # write a unit test that the Agent constructor is jittable
+        jitted_constructor = jit(construct_agent)
+        agent_instance = jitted_constructor(
+            A_batched, B_batched, pA_batched, pB_batched, D_batched
+        )
+        self.assertIsInstance(agent_instance, Agent)
+
+    def test_valid_gradients_one_step_ahead(self):
```

Comment:
At the moment I believe this test will fail at the first exception, which means we would lose coverage on the subsequent ones. It probably makes sense to break this test out into parameterized sub tests - one for each inference algorithm, with the added benefit of being able to run each version of the test in parallel.
https://docs.pytest.org/en/stable/how-to/parametrize.html
But that's something I will take a look at separately